builtin/providers/azurerm/resource_arm_loadbalancer_rule_test.go

```diff
@@ -115,6 +115,28 @@ func TestAccAzureRMLoadBalancerRule_removal(t *testing.T) {
 	})
 }
 
+// https://github.com/hashicorp/terraform/issues/9424
+func TestAccAzureRMLoadBalancerRule_inconsistentReads(t *testing.T) {
+	var lb network.LoadBalancer
+	ri := acctest.RandInt()
+	lbRuleName := fmt.Sprintf("LbRule-%s", acctest.RandStringFromCharSet(8, acctest.CharSetAlpha))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMLoadBalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLoadBalancerRule_inconsistentRead(ri, lbRuleName),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadBalancerExists("azurerm_lb.test", &lb),
+					testCheckAzureRMLoadBalancerRuleExists(lbRuleName, &lb),
+				),
```

Comment:
Given that it's important that `azurerm_lb_probe.test` and `azurerm_lb_backend_address_pool.test` aren't attached to `azurerm_lb_rule.test` so that you can create them concurrently, should you also check that `azurerm_lb_probe.test` and `azurerm_lb_backend_address_pool.test` are created correctly?